tsdb/db.go

```diff
@@ -899,22 +911,36 @@ func (db *DB) run() {
 			case db.compactc <- struct{}{}:
 			default:
 			}
+		case <-OOOScheduledCompact.C:
+      OOOScheduledCompact.Reset(db.opts.OutOfOrderCompactInterval)
+
+      db.metrics.compactionsTriggered.Inc()
+
+      db.autoCompactMtx.Lock()
+      if db.autoCompact {
+        if err := db.CompactOOOHead(); err != nil {
```

Comment:
Since we are doing OOO compaction outside of Compact() now, we need to increment `compactionsFailed` inside `CompactOOOHead()` if the compaction failed.